src/Mvc/src/Microsoft.AspNetCore.Mvc.Formatters.Xml/XmlSerializerInputFormatter.cs

```diff
@@ -159,6 +159,12 @@ public virtual InputFormatterExceptionPolicy ExceptionPolicy
             }
             // XmlSerializer wraps actual exceptions (like FormatException or XmlException) into an InvalidOperationException
             // https://github.com/dotnet/corefx/blob/master/src/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs#L652
+            catch (InvalidOperationException exception) when (exception.InnerException != null &&
```

Comment:
Can you expand upon this in the comments? You did some valuable sleuthing here and I don't think it should be lost to the sands of time.